fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java

```diff
@@ -322,6 +325,15 @@ public List<Long> createShards(int numShards, int replicaNum, FilePathInfo pathI
 
             for (int i = 0; i < numShards; ++i) {
                 builder.setShardId(GlobalStateMgr.getCurrentState().getNextId());
+                if (matchShardIds != null) {
```

Comment:
should check `matchShardIds.size() == numShards`

Reply:
done